core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryParser.java

```diff
@@ -142,7 +142,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 } else if (parseContext.parseFieldMatcher().match(currentFieldName, Fields.MINIMUM_SHOULD_MATCH)) {
                     mltQuery.setMinimumShouldMatch(parser.text());
                 } else if ("analyzer".equals(currentFieldName)) {
-                    analyzer = parseContext.analysisService().analyzer(parser.text());
+                    analyzer = parseContext.context().analysisService().analyzer(parser.text());
```

Comment:
this as well will need to be moved to the data node once we refactor this query?